### PR TITLE
Add API to codeowners

### PR DESCRIPTION
**Pull request by caniszczyk**
Enforce Harvey to review any API changes.

Signed-off-by: Chris Aniszczyk <caniszczyk@gmail.com>

### CODE REVIEW

File: CODEOWNERS [hunk 1]
```diff
@@ -2,6 +2,8 @@
 # By default, @envoyproxy/maintainers own everything.
 #*       @envoyproxy/maintainers
 
+# api
+/api/* @htuch
```
Comment (htuch) [inline CODEOWNERS]: Do you need trailing `/*`?

Comment (htuch) [reply CODEOWNERS]: I think this should be `/api/`

Comment (caniszczyk) [reply CODEOWNERS]: fixed

**Comment by mattklein123** [comment]
@htuch I think we need to discuss this further. I don't think we can block on a single person doing reviews. If we want there to be a smaller set of approved API reviewers that's fine, but I think we should discuss before making this a requirement. cc @envoyproxy/maintainers

**Comment by jmarantz** [comment]
+1 -- why not the senior maintainers, or at least a minimum of 3 of them? Harvey can be the primary POC.

**Comment by htuch** [comment]
I'm happy to discuss. This is a followup to the earlier discussion around how we get to the point where we are able to enforce API stability and a cleaner API design. I'd like a dedicated team to be created of people who are committed to API consistency and are part of the UDPA effort. I'd like to have us all on the same page about what constitutes good API practices. If we can create such an @-group then let's switch to this.

**Comment by htuch** [comment]
FWIW, the design document bullet point covering this is https://docs.google.com/document/d/1xeVvJ6KjFBkNjVspPbY_PwEDHC7XPi0J5p1SqUXcCl8/edit#bookmark=id.xaolar1o4c32

**Comment by mattklein123** [comment]
@htuch 100% agreed on having a set of API reviewers, I'm mainly balking at you being the only reviewer given vacations, etc.

**Comment by htuch** [comment]
@envoyproxy/api-shepherds created and switching to this group as owners in https://github.com/envoyproxy/envoy/pull/7121.

**Pull request merged**